### PR TITLE
Add scrolling ability to AcceptDialog.

### LINKED ISSUE #76253: File overwrite confirmation dialogue can be taller than screen
### Godot version

4.0.2.stable.official.7a0977ce2

### System information

Windows 11, NVIDIA - NVIDIA GeForce RTX 3070 Ti

### Issue description

I have two folders with identical files - about 50 files in total. If I drag all the files from folder A to B, I get a popup asking me to confirm the overwrite. Problem is the confirmation window lists all the file names to be overwritten and because there's so many, I can't actually see the confirm or cancel buttons at the bottom of the popup. The popup isn't scrollable and I can't resize it to make it smaller vertically.

image

### Steps to reproduce

Have a folder 'foo' with 50 files in it. Doesn't matter what the files are. Empty resources will work.
Duplicate the folder and name the duplicate 'bar'.
In the FileSystem drag all the files in 'foo' into 'bar'.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by nongvantinh**
Fixes: #76253

### CODE REVIEW

File: scene/gui/dialogs.cpp [hunk 1]
```diff
@@ -395,12 +395,15 @@ AcceptDialog::AcceptDialog() {
 	bg_panel = memnew(Panel);
 	add_child(bg_panel, false, INTERNAL_MODE_FRONT);
 
+	scroll_bar = memnew(ScrollContainer);
```
Comment (EricEzaM) [inline scene/gui/dialogs.cpp:398]: Calling it scroll bar but instantiating a scroll container probably isn't a good idea, considering ScrollBar is a call that exists jn the API. I recommend updating this member name.

**Comment by YuriSizov** [comment]
I don't think we need this for the built-in label. The built-in label should only be used for short messages, which don't require scrolling or complex formatting. In most cases the dialog should also just update to the size of the message, and the scroll container makes it impossible.

Instead, the dialog in question from the linked PR should be remade with a custom layout instead of relying on the built-in label.

**Comment by YuriSizov** [comment]
My suggestion is to find the exact dialog that is used in the issue, and instead of setting its contents as text create a custom layout for that specific dialog. It would probably be something like

```
- VBox
  - Label (for the overall description)
  - Tree/ItemList (for the list of files)
```

Can also use a combination of a ScrollContainer and a Label for the list, if you want to.

**Comment by nongvantinh** [comment]
I have tried three different approaches:

1. Wrapping the `message_label` inside the `ScrollContainer` so that its content does not cause the Dialog to overflow the screen. The advantage of this approach is that it requires minimal changes. However, it has the downside that the Dialog does not resize automatically based on the `message_label`'s value.

2. Creating a `VboxContainer` to wrap the `message_label` and a `Tree` node. This solution works and automatically adds scrolling ability when the number of `TreeItem`s presented does not fit the Dialog. However, it suffers from the same problem as approach 1 in that the Dialog does not resize automatically based on the number of `TreeItem`.

3. Creating a `VboxContainer` to wrap the `message_label` and an `ItemList` node. This has the same effect as method 2, but now I have to connect two signals (canceled and confirmed) of the Dialog to know when to clear the `ItemList` to make it ready for the next popup.

I also tried using `set_auto_height` for `ItemList` to make the Dialog auto-resizing, but it caused the Dialog to overflow the screen.

Based on these results, I consider the first approach using `ScrollContainer` to be the best option. It just needs a modification to make it expand as the `message_label`'s value changes up to a certain boundary size. What do you think about this?

**Comment by KoBeWi** [comment]
I think the dialog just needs a minimum height.

**Comment by nongvantinh** [comment]
After some experimentation, I decided to use `Tree` instead of `ItemList` because in the future, we may need to display the location of files with more detail by showing the tree where they are rooted, instead of just displaying their path.

Regarding the current pull request, it does not automatically update the size of the Dialog based on the children of its `contents`. In the past, it relied on the size of the `message_label` to determine the size of the Dialog. However, if it is acceptable, I can implement the behavior to update the Dialog size based on the `contents`'s children.

**Comment by YuriSizov** [comment]
Superseded by https://github.com/godotengine/godot/pull/77179. Thanks for your contribution nonetheless!

**Pull request closed without merging**